itest/lnd_route_blinding.go

```diff
@@ -0,0 +1,290 @@
+				// expected to be the introduction node.
+				BlindedNode:   carolBlindedBytes,
+				EncryptedData: encryptedDataCarol,
+			},
+			{
+				BlindedNode:   blinded1Bytes,
+				EncryptedData: encryptedData1,
+			},
+			{
+				BlindedNode:   blinded2Bytes,
+				EncryptedData: encryptedData2,
+			},
+		},
+	}
+
+	// Create a blinded payment that has aggregate cltv and fee params
+	// for our route.
+	var (
+		blindedBaseFee   uint64 = 1500
+		blindedCltvDelta uint32 = 125
+	)
+
+	blindedPayment := &lnrpc.BlindedPaymentPath{
+		BlindedPath:    route,
+		BaseFeeMsat:    blindedBaseFee,
+		TotalCltvDelta: blindedCltvDelta,
+	}
+
+	// Query for a route to the blinded path constructed above.
+	ctxt, cancel = context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	var (
+		paymentAmt int64 = 100_000
+	)
+	req := &lnrpc.QueryRoutesRequest{
+		AmtMsat: paymentAmt,
+		BlindedPaymentPaths: []*lnrpc.BlindedPaymentPath{
+			blindedPayment,
+		},
+	}
+
+	resp, err := alice.RPC.LN.QueryRoutes(ctxt, req)
+	require.NoError(ht, err)
+	require.Len(ht, resp.Routes, 1)
+
+	// Payment amount and cltv will be included for the bob/carol edge
+	// (because we apply on the outgoing hop), and the blinded portion of
+	// the route.
+	totalFee := bobCarolBase + blindedBaseFee
+	totalAmt := uint64(paymentAmt) + totalFee
+	totalCltv := info.BlockHeight + bobCarolDelta + blindedCltvDelta
+
+	// Alice -> Bob
+	//   Forward: total - bob carol fees
+	//   Expiry: total - bob carol delta
+	//
+	// Bob -> Carol
+	//  Forward: 101500 (total + blinded fees)
+	//  Expiry: Height + final delta
+	//  Encrypted Data: enc_carol
+	//
+	// Carol -> Blinded 1
+	//  Forward/ Expiry: 0
+	//  Encrypted Data: enc_1
+	//
+	// Blinded 1 -> Blinded 2
+	//  Forward/ Expiry: Height
+	//  Encrypted Data: enc_2
+	hop0Amount := int64(totalAmt - bobCarolBase)
+	hop0Expiry := totalCltv - bobCarolDelta
+	finalHopExpiry := totalCltv - bobCarolDelta - blindedCltvDelta
+
+	expectedRoute := &lnrpc.Route{
+		TotalTimeLock: totalCltv,
+		TotalAmtMsat:  int64(totalAmt),
+		TotalFeesMsat: int64(totalFee),
+		Hops: []*lnrpc.Hop{
+			{
+				ChanId:           aliceBobChan.ChanId,
+				Expiry:           hop0Expiry,
+				AmtToForwardMsat: hop0Amount,
+				FeeMsat:          int64(bobCarolBase),
+				PubKey:           bob.PubKeyStr,
+			},
+			{
+				ChanId:        bobCarolChan.ChanId,
+				PubKey:        carol.PubKeyStr,
+				BlindingPoint: blindingBytes,
+				FeeMsat:       int64(blindedBaseFee),
+				EncryptedData: encryptedDataCarol,
+			},
+			{
+				PubKey: hex.EncodeToString(
+					blinded1Bytes,
+				),
+				EncryptedData: encryptedData1,
+			},
+			{
+				PubKey: hex.EncodeToString(
+					blinded2Bytes,
+				),
+				AmtToForwardMsat: paymentAmt,
+				Expiry:           finalHopExpiry,
+				EncryptedData:    encryptedData2,
+				TotalAmtMsat:     uint64(paymentAmt),
+			},
+		},
+	}
+
+	r := resp.Routes[0]
+	assert.Equal(ht, expectedRoute.TotalTimeLock, r.TotalTimeLock)
```

Comment:
use `require`?

Reply:
For this test I've found it more useful to assert on the values so that I can see _everything_ that went wrong (rather than killing the test on the first bad value) because it gives a clearer indication of why it's failing. 

Happy to change, but I do think it's valuable to `assert` sometimes.

Reply:
oh nice I didn't know that! Yeah that's very helpful.